movai_core_shared/core/zmq/zmq_manager.py

```diff
@@ -0,0 +1,85 @@
+from logging import getLogger
+
+from movai_core_shared.core.zmq.zmq_base import ZMQBase
+from movai_core_shared.core.zmq.zmq_client import ZMQClient, AsyncZMQClient
+from movai_core_shared.core.zmq.zmq_subscriber import ZMQSubscriber, AsyncZMQSubscriber
+from movai_core_shared.core.zmq.zmq_publisher import ZMQPublisher, AsyncZMQPublisher
+from movai_core_shared.core.zmq.zmq_helpers import generate_zmq_identity
+from movai_core_shared.exceptions import ArgumentError
+
+
+class ZMQManager:
+    _logger = getLogger("ZMQManager")
+    _clients = {}
+    _async_clients = {}
+    _publishers = {}
+    _async_publishers = {}
+    _subscribers = {}
+    _async_subscribers = {}
+
+    zmq_types = {
+        ZMQClient.__name__: "dealer",
+        AsyncZMQClient.__name__: "dealer",
+        ZMQPublisher.__name__: "pub",
+        AsyncZMQPublisher.__name__: "pub",
+        ZMQSubscriber.__name__: "sub",
+        AsyncZMQSubscriber.__name__: "sub"
+    }
+    
+    @classmethod
+    def validate_server_addr(cls, server_addr: str):
+        if not isinstance(server_addr, str):
+            error_msg = "The argument server_addr must be of type string"
+            cls._logger.error(error_msg)
+            raise ArgumentError(error_msg)
+
+    def validate_zmq_type(cls, zmq_object: ZMQBase):
+        if not isinstance(zmq_object, ZMQBase):
+            error_msg = f"The argument zmq_object must be of type {ZMQBase.__class__.__name__}"
+            cls._logger.error(error_msg)
+            raise ArgumentError(error_msg)
+
+    @classmethod
+    def _get_zmq_object(cls, server_addr: str, object_type: ZMQBase, objects_dict: dict) -> ZMQBase:
+        cls.validate_server_addr(server_addr)
+        
+        if server_addr in objects_dict:
+            return objects_dict[server_addr]
+        else:
+            if object_type.__class__.__name__ in cls.zmq_types:
+                identity = cls.zmq_types[object_type.__class__.__name__]
+            else:
+                identity = generate_zmq_identity("")
+            zmq_object = object_type(identity, server_addr)
+            objects_dict[server_addr] = zmq_object
+            return zmq_object
+
+    @classmethod
+    def get_client(cls, server_addr: str) -> ZMQClient:
```

Comment:
instead of all of those we can create an enum to represent the different types
send this enum to the main function "get_zmq_object" 
for example we could have: 
```
from enum import Enum 
class ZMQType(Enum):
    Client = 1
    AsyncClient = 2
    Sub = 3
    AsyncSub = 4
    Pub = 5
    AsyncPub = 6
```
and when we use it will make sense
```
ZMQManager.get_zmq_object(server_addr, ZMQType.AsyncPub)
```